sdk-api-src/content/handleapi/nf-handleapi-duplicatehandle.md

```diff
@@ -342,16 +358,24 @@ The <i>dwDesiredAccess</i> parameter specifies the new handle's access rights. A
 <a href="https://docs.microsoft.com/windows/desktop/winstation/window-station-security-and-access-rights">Window-Station Security and Access Rights</a>
 </li>
 </ul>
-In some cases, the new handle can have more access rights than the original handle. However, in other cases, 
-<b>DuplicateHandle</b> cannot create a handle with more access rights than the original. For example, a file handle created with the GENERIC_READ access right cannot be duplicated so that it has both the GENERIC_READ and GENERIC_WRITE access right.
 
-Normally the target process closes a duplicated handle when that process is finished using the handle. To close a duplicated handle from the source process,  call <b>DuplicateHandle</b> with the following parameters: 
+In some cases, the duplicate handle can be granted more access rights than the original handle, but in other cases it cannot. For example, process handles can be duplicated with <b>PROCESS_ALL_ACCESS</b> rights regardless of the original handle's access rights, whereas a file handle created with the <b>GENERIC_READ</b> access right cannot be duplicated so that it has the <b>GENERIC_WRITE</b> access right.
+
```

Comment:
We really should spell out the details of this. It applies to the COM/RPC/ALPC marshaling as well, and sadly is about as clear as mud, even internally. In what scenarios will it work, in what scenarios will it silently be ignored, and in what scenarios will it fail. When dealing with a middle layer library you may not be the one opening the handle and can't make guarantees about the permissions it has, but want to ensure the target has at least X and NEVER Y. Clearly, there are rules on this sort of thing (and hopefully tests that validate them) we've just never done the hard work of collecting all the details and documenting it. Which is pretty bad from something security related.

Reply:
Agreed, but that seems like a Very Big Project and would be a separate topic (then we can refer to it from this page). Do we even know where to start? We had a hard time even figuring out which kinds of handles CAN be duplicated (never really resolved).